project/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,133 @@
+import java.net.SocketException;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+
+/**
+ * A tester for the data servlet.
+ */
+@RunWith(JUnit4.class)
+public class DataServletTest {
+  // JSON response can be referenced below: @see <a href=
+  // "https://developers.google.com/civic-information/docs/using_api#voterinfoquery-response:">
+  // response structure</a>.
+  private static final String RELEVANT_ADDRESS_ELECTION_RESPONSE =
+      "{" +
+      "  contests: []," +
+      "  state: [" +
+      "    {" +
+      "      sources: [" +
+      "        {" +
+      "          name: \"Voting Information Project\"" +
+      "        }" +
+      "      ]" +
+      "    }" +
+      "  ]" +
+      "}";
+  private static final String RELEVANT_NONSPECIFIC_ADDRESS_ELECTION_RESPONSE =
+      "{" +
+      "  state: [" +
+      "    {" +
+      "      sources: [" +
+      "        {" +
+      "          name: \"Voting Information Project\"" +
+      "        }" +
+      "      ]" +
+      "    }" +
+      "  ]" +
+      "}";
+  private static final String IRRELEVANT_ADDRESS_ELECTION_RESPONSE =
+      "{" +
+      "  state: [" +
+      "    {" +
+      "      sources: [" +
+      "        {" +
+      "          name: \"\"" +
+      "        }" +
+      "      ]" +
+      "    }" +
+      "  ]" +
+      "}";
+  private static final DataServlet dataServlet = new DataServlet();
+
+  @Test
+  public void requestHttpAndBuildJsonResponseFromCivicInformation_returnResponseStringAsIs()
+      throws IOException, SocketException {
+    // Query the Civic Information API with a mock HTTP client + mock callback function of type
+    // {@code ResponseHandler<String>} that converts any {@code HttpResponse} response to {@code
+    // RELEVANT_ADDRESS_ELECTION_RESPONSE}. {@code httpGet} is irrelevant in this test. The
+    // method should return the response as is without changing it.
+    CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+    HttpGet httpGet = new HttpGet("");
+    ArgumentCaptor<ResponseHandler<String>> argumentCaptor =
+        ArgumentCaptor.forClass(ResponseHandler.class);
+    when(httpClient.execute(anyObject(), argumentCaptor.capture()))
+        .thenReturn(RELEVANT_ADDRESS_ELECTION_RESPONSE);
+    String responseBody =
+        dataServlet.requestHttpAndBuildJsonResponseFromCivicInformation(httpClient, httpGet);
+    assertThat(responseBody).isEqualTo(RELEVANT_ADDRESS_ELECTION_RESPONSE);
```

Comment:
use assertEquals

Reply:
@suparnabarua  Actually, this is the Google Java Truth syntax, so these `assertThat`s are correct: https://truth.dev/

Reply:
(Google Java Truth provides better automatic assertion error messaging.)

Reply:
TIL. Thanks